packages/nodes-base/src/NodeVersionedType.ts

```diff
@@ -0,0 +1,21 @@
+import { INodeType, INodeTypeBaseDescription, INodeVersionedType, INodeVersions } from 'n8n-workflow';
+
+export class NodeVersionedType implements INodeVersionedType {
+	defaultVersion: number;
+	nodeVersions: INodeVersions;
+	description: INodeTypeBaseDescription;
+
+	constructor(nodeVersions: INodeVersions, description: INodeTypeBaseDescription, defaultVersion?: number) {
+		this.nodeVersions = nodeVersions;
+		this.defaultVersion = defaultVersion || Math.max(...Object.keys(nodeVersions).map(k => parseInt(k, 10)));
```

Comment:
1. Ideally `??` instead of `||`.
2. Another option: `.map(Number)`
3. The keys of `nodesVersions` are already numbers. Why convert them into numbers?
4. Since `defaultVersion` can be the default version or the latest version, maybe `currentVersion` would be a more fitting key name.